lib/authCodeGrant.js

```diff
@@ -182,14 +215,47 @@ function saveAuthCode (done) {
 }
 
 /**
+ * Save access token with model
+ *
+ * @param  {Function} done
+ * @this   OAuth
+ */
+function saveAccessToken (done) {
+  var accessToken = this.accessToken;
+
+  // Object idicates a reissue
+  if (typeof accessToken === 'object' && accessToken.accessToken) {
+    this.accessToken = accessToken.accessToken;
+    return done();
+  }
+
+  var expires = null;
+  if (this.config.accessTokenLifetime !== null) {
+    expires = new Date();
+    expires.setSeconds(expires.getSeconds() + this.config.accessTokenLifetime);
+  }
+
+  this.model.saveAccessToken(accessToken, this.client.clientId, expires,
+      this.user, function (err) {
+    if (err) return done(error('server_error', false, err));
+    done();
+  });
+}
+
+/**
  * Check client against model
  *
  * @param  {Function} done
  * @this   OAuth
  */
 function redirect (done) {
-  this.res.redirect(this.client.redirectUri + '?code=' + this.authCode +
-      (this.req.query.state ? '&state=' + this.req.query.state : ''));
+  if(this.responseType === 'code') {
+    this.res.redirect(this.client.redirectUri + '?code=' + this.authCode +
+        (this.req.query.state ? '&state=' + this.req.query.state : ''));
+  } else {
+    this.res.redirect(this.client.redirectUri + '#token=' + this.accessToken +
+        (this.req.query.state ? '#token=' + this.req.query.state : ''));
```

Comment:
Yes, but you have `token` twice. After the check for the `state` query variable you should be adding state to the fragment, not token. 






On Friday, Aug 14, 2015 at 4:00 AM, Jonatan Männchen <notifications@github.com>, wrote:

In lib/authCodeGrant.js:

>   * Check client against model
>   *
>   * @param  {Function} done
>   * @this   OAuth
>   */
>  function redirect (done) {
> -  this.res.redirect(this.client.redirectUri + '?code=' + this.authCode +
> -      (this.req.query.state ? '&state=' + this.req.query.state : ''));
> +  if(this.responseType === 'code') {
> +    this.res.redirect(this.client.redirectUri + '?code=' + this.authCode +
> +        (this.req.query.state ? '&state=' + this.req.query.state : ''));
> +  } else {
> +    this.res.redirect(this.client.redirectUri + '#token=' + this.accessToken +
> +        (this.req.query.state ? '#token=' + this.req.query.state : ''));

I don't. The #token is important since I want to pass it on to a client. (Not the server)

I implemented it while looking at this article:
https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2#grant-type-implicit


—
Reply to this email directly or view it on GitHub.